app/models/season/phase_switcher.rb

```diff
@@ -1,44 +1,60 @@
+# frozen_string_literal: true
+
 class Season::PhaseSwitcher
 
+  PHASES = %i(
+    fake_proposals_phase
+    fake_application_phase
+    fake_coding_phase
+    back_to_reality
+    ).freeze
+
   def self.season
     Season.current
   end
 
+  def self.destined(phase)
+    raise ArgumentError.new("#{phase} is not a valid phase") unless phase.in? PHASES
+    self.public_send(phase)
+  end
+
+  private
```

Comment:
Alright. Just out of curiosity, why do you want to have all methods on `self`? I could also have imagined this class being a service object or a "fake singleton" using `class << self`.